lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -658,7 +658,26 @@ sub overview {
         groups  => $groups,
         until   => $until,
     );
-    my @latest_jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args)->latest_jobs($until);
+
+    my @latest_jobs;
+    my $group_ids = $search_args->{groupids};
+    if (defined $group_ids && @$group_ids > 1) {
+        my @composed_search_arch = map {
+            my $local_search_args = {
+                group_id => $_,
+                build    => $self->schema->resultset('Jobs')->latest_build((groupid => $_))};
+            $local_search_args->{distri}  = $search_args->{distri}  if (defined($search_args->{distri}));
+            $local_search_args->{version} = $search_args->{version} if (defined($search_args->{version}));
+            $local_search_args;
+        } @$group_ids;
+
+        @latest_jobs
+          = $self->schema->resultset('Jobs')->search(\@composed_search_arch)->latest_jobs_multiple_groups($until);
+    }
+    else {
+        @latest_jobs = $self->schema->resultset('Jobs')->complex_query(%$search_args)->latest_jobs($until);
+    }
```

Comment:
Instead of making the sub "overview" which is already big even bigger please extract some functions